### ISSUE TITLE
Brackets in the ID is not supported

### ISSUE THREAD

**Issue by balexandre** [qwertypants/jQuery-Word-and-Character-Counter-Plugin#9]
To use auto inject `IEnumerable` the id's need to be of the format `Listname[ index ]_PropertyName` and the name has to use a `.` instead of the `_` to separate the list variable from it's property.

example for those not familiarize with C# MVC:

    public class Cars {
      void Cars(string brand, string model) {
        this.Brand = brand; this.Model = model;
      }
      public string Brand { get; set; }
      public string Model { get; set; }
    }

    List list = new List<Cars>();
    list.Add("Audi", "RS6");
    list.Add("BMW", "M6");

when looping in a MVC View, this is the *view code*

    <ul>
      @for(int iCar = 0; i < Model.Count, i++) {
        var car = Model[i];
        <li>
          
          <textarea id="Cars[@(i)].Brand" name="Cars[@(i)].Brand">@car.Brand</textarea>
          <textarea id="Cars[@(i)].Model" name="Cars[@(i)].Model">@car.Model</textarea>
        </li>
      }
    </ul>

the generated HTML is:

    <ul>
        <li>
          
          <textarea id="Cars[0]_Brand" name="Cars[0].Brand">Audi</textarea>
          <textarea id="Cars[0]_Model" name="Cars[0].Model">RS6</textarea>
        </li>
        <li>
          
          <textarea id="Cars[1]_Brand" name="Cars[1].Brand">BMW</textarea>
          <textarea id="Cars[1]_Model" name="Cars[1].Model">M6</textarea>
        </li>
    </ul>

this makes a form to be submitted and auto-inject the new values to the same object collection

Bottom line is that if you have an `id` of this type, this plugin will not work at all. It writes the `msg` but **will not write the counter or throw any exception**.

Live Example on JSBin: http://jsbin.com/nifamu/2/edit?html,js,output

**Comment by qwertypants** [comment]
Thanks for your feedback. 

Try experimenting with different selectors. When using brackets in an idea, you'll need to preface it with ```\\``` http://api.jquery.com/category/selectors/.

**Comment by balexandre** [comment]
But that should be the plugin task to avoid such issue, brackets are HTML, and if I change them, I will never get the behavior I need on my own codding language.

**Comment by qwertypants** [comment]
I don't see any value in adding this to the plugin, as it creates overhead and can be avoided by using `\\` or other valid jQuery selectors.
